src/providers/postgres/qgspostgresprovider.cpp

```diff
@@ -622,14 +641,41 @@ QString QgsPostgresUtils::whereClause( const QgsFeatureIds &featureIds, const Qg
         const auto constFeatureIds = featureIds;
         for ( const QgsFeatureId featureId : constFeatureIds )
         {
-          expr += delim + FID_TO_STRING( ( pkType == PktOid ? featureId : pkType == PktUint64 ? featureId : FID2PKINT( featureId ) ) );
+          expr += delim + FID_TO_STRING( ( pkType == PktOid ? featureId : FID2PKINT( featureId ) ) );
           delim = ',';
         }
         expr += ')';
       }
 
       return expr;
     }
+    case PktInt64:
+    case PktUint64:
+    {
+      QString expr;
+
+      //simple primary key, so prefer to use an "IN (...)" query. These are much faster then multiple chained ...OR... clauses
+      if ( !featureIds.isEmpty() )
+      {
+        QString delim;
+        expr = QStringLiteral( "%1 IN (" ).arg( QgsPostgresConn::quotedIdentifier( fields.at( pkAttrs[0] ).name() ) );
+
+        const auto constFeatureIds = featureIds;
+        for ( const QgsFeatureId featureId : constFeatureIds )
```

Comment:
Please use this style (the other approach was introduced by an automatic conversion)

```suggestion
        for ( const QgsFeatureId featureId : qgis::as_const( featureIds ) )
```

Reply:
May I change this on other parts of the code? Those that handle PktFidMap and PktInt?

Reply:
Done, see 9607570 (also changed the other parts of the code)